crates/cubecl-runtime/src/memory_management/memory_lock.rs

```diff
@@ -6,7 +9,10 @@ use crate::storage::StorageId;
 /// used for allocations currently.
 #[derive(Debug, Default)]
 pub struct MemoryLock {
+    #[cfg(feature = "std")]
     locked: HashSet<StorageId>,
+    #[cfg(not(feature = "std"))]
+    locked: BTreeSet<StorageId>,
```

Comment:
We should always use BTreeSet I think, simpler and storage id is pretty much just a number.